core/src/main/java/com/github/maracas/visitors/ConstructorRemovedVisitor.java

```diff
@@ -0,0 +1,67 @@
+package com.github.maracas.visitors;
+
+import com.github.maracas.detection.APIUse;
+
+import japicmp.model.JApiCompatibilityChange;
+import spoon.reflect.code.CtConstructorCall;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.code.CtNewClass;
+import spoon.reflect.code.CtSuperAccess;
+import spoon.reflect.reference.CtExecutableReference;
+import spoon.reflect.reference.CtTypeReference;
+
+/**
+ * Visitor in charge of gathering all constructor
+ * removed issues in client code.
+ * <p>
+ * CONSTRUCTOR_REMOVED detected cases:
+ * <ul>
+ * <li> Any call to the removed constructor.
+ *      Example: <pre>new TypeName();</pre>
+ * <li> Call to the removed constructor when creating
+ *      an anonymous class.
+ *      Example: <pre>new AnonymClass() { }</pre>
+ * <li> Invocation to super() constructor.
+ *      Example: <pre>super()</pre>
+ * </ul>
+ */
+public class ConstructorRemovedVisitor extends BreakingChangeVisitor {
+
+	private final CtExecutableReference<?> mRef;
+
+	public ConstructorRemovedVisitor(CtExecutableReference<?> mRef) {
+		super(JApiCompatibilityChange.CONSTRUCTOR_REMOVED);
+		this.mRef = mRef;
+	}
+
+	@Override
+	public <T> void visitCtConstructorCall(CtConstructorCall<T> consCall) {
+		if (mRef.equals(consCall.getExecutable())) {
+			detection(consCall, consCall.getExecutable(), mRef, APIUse.INSTANTIATION);
+		}
+	}
+
+	@Override
+	public <T> void visitCtNewClass(CtNewClass<T> anonymClass) {
+		CtTypeReference<?> superclass = anonymClass.getAnonymousClass().getSuperclass();
+
+		if (superclass != null) {
+			superclass.getAllExecutables().forEach(c -> {
```

Comment:
Is it necessary to visit _all_ super executables? This has a _huge_ impact on performance (x10); though I'm not sure why Spoon gets so slow in this case. Would something like that work instead?

```java
if (superclass != null)
			superclass.getDeclaredExecutables()
				.stream()
				.filter(exec -> mRef.equals(exec))
				.forEach(exec ->
					detection(anonymClass, exec, mRef, APIUse.INSTANTIATION)
				);
```

Reply:
You are completely right. I see that you already modified it. Thanks :)